### PR TITLE
[spritekit] Update for Xcode 9 beta 1, 2 & 3

### PR DESCRIPTION
**Pull request by VincentDondain**
- Added Quaternion support to XI runtime (simd_quatf).

### CODE REVIEW

File: Make.config [hunk 1]
```diff
@@ -60,7 +60,7 @@ MIN_MONO_URL=https://bosstoragemirror.blob.core.windows.net/wrench/mono-2017-04/
 # Minimum Visual Studio version
 MIN_VISUAL_STUDIO_URL=https://bosstoragemirror.blob.core.windows.net/wrench/monodevelop-lion-dogfood-vNext/8f/8f1c13cb983138ee63bd53e09908ea5e737988cd/VisualStudioForMac-Preview-7.0.0.2728.dmg
 MIN_VISUAL_STUDIO_VERSION=7.0.0.2728
-MAX_VISUAL_STUDIO_VERSION=7.2.99
+MAX_VISUAL_STUDIO_VERSION=7.3.99
```
Comment (dalexsoto) [inline Make.config]: Why is this required? I think we should keep the same as 15.3 unless we **require** this

File: src/spritekit.cs [hunk 2]
```diff
@@ -1954,28 +1991,28 @@ partial interface SKAction : NSCoding, NSCopying {
 		
 		// These are in a category
 		[Static, Export ("moveByX:y:duration:")]
-		SKAction MoveBy (nfloat deltaX, nfloat deltaY, double sec);
+		SKAction MoveBy (nfloat deltaX, nfloat deltaY, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: @spouliot is this a breaking change? I am in favor of the change if we can allow it but @VincentDondain  I would use `durationInSeconds` instead if seconds is the unit of measure of duration

File: src/spritekit.cs [hunk 3]
```diff
@@ -1954,28 +1991,28 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("moveBy:duration:")]
 		SKAction MoveBy (CGVector delta, double duration);
 
 		[Static, Export ("moveTo:duration:")]
-		SKAction MoveTo (CGPoint location, double sec);
+		SKAction MoveTo (CGPoint location, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 4]
```diff
@@ -1954,28 +1991,28 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("moveToX:duration:")]
-		SKAction MoveToX (nfloat x, double sec);
+		SKAction MoveToX (nfloat x, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 5]
```diff
@@ -1954,28 +1991,28 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("moveToY:duration:")]
-		SKAction MoveToY (nfloat y, double sec);
+		SKAction MoveToY (nfloat y, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 6]
```diff
@@ -1954,28 +1991,28 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("rotateByAngle:duration:")]
-		SKAction RotateByAngle (nfloat radians, double sec);
+		SKAction RotateByAngle (nfloat radians, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 7]
```diff
@@ -1954,28 +1991,28 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("rotateToAngle:duration:")]
-		SKAction RotateToAngle (nfloat radians, double sec);
+		SKAction RotateToAngle (nfloat radians, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 8]
```diff
@@ -1954,28 +1991,28 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("rotateToAngle:duration:shortestUnitArc:")]
-		SKAction RotateToAngle (nfloat radians, double sec, bool shortedUnitArc);
+		SKAction RotateToAngle (nfloat radians, double duration, bool shortedUnitArc);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 9]
```diff
@@ -1990,28 +2027,28 @@ partial interface SKAction : NSCoding, NSCopying {
 		SKAction ResizeToHeight (nfloat height, double duration);
 
 		[Static, Export ("scaleBy:duration:")]
-		SKAction ScaleBy (nfloat scale, double sec);
+		SKAction ScaleBy (nfloat scale, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above, what is the unit of measure? -> `durationInSeconds` maybe?

File: src/spritekit.cs [hunk 10]
```diff
@@ -1990,28 +2027,28 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("scaleXBy:y:duration:")]
-		SKAction ScaleBy (nfloat xScale, nfloat yScale, double sec);
+		SKAction ScaleBy (nfloat xScale, nfloat yScale, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 11]
```diff
@@ -1990,28 +2027,28 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("scaleTo:duration:")]
-		SKAction ScaleTo (nfloat scale, double sec);
+		SKAction ScaleTo (nfloat scale, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 12]
```diff
@@ -1990,28 +2027,28 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("scaleXTo:y:duration:")]
-		SKAction ScaleTo (nfloat xScale, nfloat yScale, double sec);
+		SKAction ScaleTo (nfloat xScale, nfloat yScale, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 13]
```diff
@@ -1990,28 +2027,28 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("scaleXTo:duration:")]
-		SKAction ScaleXTo (nfloat scale, double sec);
+		SKAction ScaleXTo (nfloat scale, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 14]
```diff
@@ -1990,28 +2027,28 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("scaleYTo:duration:")]
-		SKAction ScaleYTo (nfloat scale, double sec);
+		SKAction ScaleYTo (nfloat scale, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 15]
```diff
@@ -1990,28 +2027,28 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[iOS (10,0)][Mac (10,12)]
 		[TV (10,0)]
 		[Static]
 		[Export ("scaleToSize:duration:")]
-		SKAction ScaleTo (CGSize size, double sec);
+		SKAction ScaleTo (CGSize size, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 16]
```diff
@@ -2026,16 +2063,16 @@ partial interface SKAction : NSCoding, NSCopying {
 		SKAction RepeatActionForever (SKAction action);
 
 		[Static, Export ("fadeInWithDuration:")]
-		SKAction FadeInWithDuration (double sec);
+		SKAction FadeInWithDuration (double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 17]
```diff
@@ -2026,16 +2063,16 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("fadeOutWithDuration:")]
-		SKAction FadeOutWithDuration (double sec);
+		SKAction FadeOutWithDuration (double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 18]
```diff
@@ -2026,16 +2063,16 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("fadeAlphaBy:duration:")]
-		SKAction FadeAlphaBy (nfloat factor, double sec);
+		SKAction FadeAlphaBy (nfloat factor, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 19]
```diff
@@ -2026,16 +2063,16 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("fadeAlphaTo:duration:")]
-		SKAction FadeAlphaTo (nfloat alpha, double sec);
+		SKAction FadeAlphaTo (nfloat alpha, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 20]
```diff
@@ -2055,16 +2092,16 @@ partial interface SKAction : NSCoding, NSCopying {
 		SKAction PlaySoundFileNamed (string soundFile, bool wait);
 
 		[Static, Export ("colorizeWithColor:colorBlendFactor:duration:")]
-		SKAction ColorizeWithColor (UIColor color, nfloat colorBlendFactor, double sec);
+		SKAction ColorizeWithColor (UIColor color, nfloat colorBlendFactor, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 21]
```diff
@@ -2055,16 +2092,16 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("colorizeWithColorBlendFactor:duration:")]
-		SKAction ColorizeWithColorBlendFactor (nfloat colorBlendFactor, double sec);
+		SKAction ColorizeWithColorBlendFactor (nfloat colorBlendFactor, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 22]
```diff
@@ -2055,16 +2092,16 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("followPath:duration:")]
-		SKAction FollowPath (CGPath path, double sec);
+		SKAction FollowPath (CGPath path, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 23]
```diff
@@ -2055,16 +2092,16 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("followPath:asOffset:orientToPath:duration:")]
-		SKAction FollowPath (CGPath path, bool offset, bool orient, double sec);
+		SKAction FollowPath (CGPath path, bool offset, bool orient, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 24]
```diff
@@ -2075,16 +2112,16 @@ partial interface SKAction : NSCoding, NSCopying {
 		SKAction FollowPath (CGPath path, bool offset, bool orient, nfloat speed);
 
 		[Static, Export ("speedBy:duration:")]
-		SKAction SpeedBy (nfloat speed, double sec);
+		SKAction SpeedBy (nfloat speed, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 25]
```diff
@@ -2075,16 +2112,16 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("speedTo:duration:")]
-		SKAction SpeedTo (nfloat speed, double sec);
+		SKAction SpeedTo (nfloat speed, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 26]
```diff
@@ -2075,16 +2112,16 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("waitForDuration:")]
-		SKAction WaitForDuration (double sec);
+		SKAction WaitForDuration (double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 27]
```diff
@@ -2075,16 +2112,16 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[Static, Export ("waitForDuration:withRange:")]
-		SKAction WaitForDuration (double sec, double durationRange);
+		SKAction WaitForDuration (double duration, double durationRange);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 28]
```diff
@@ -2102,7 +2139,7 @@ partial interface SKAction : NSCoding, NSCopying {
 		SKAction RunAction (SKAction action, string name);
 
 		[Static, Export ("customActionWithDuration:actionBlock:")]
-		SKAction CustomActionWithDuration (double seconds, SKActionDurationHandler actionHandler);
+		SKAction CustomActionWithDuration (double duration, SKActionDurationHandler actionHandler);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 29]
```diff
@@ -2117,27 +2154,27 @@ partial interface SKAction : NSCoding, NSCopying {
 
 		[iOS (8,0), Mac(10,10)]
 		[Static, Export ("reachTo:rootNode:duration:")]
-		SKAction ReachTo (CGPoint position, SKNode rootNode, double secs);
+		SKAction ReachTo (CGPoint position, SKNode rootNode, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 30]
```diff
@@ -2117,27 +2154,27 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[iOS (8,0), Mac(10,10)]
 		[Static, Export ("reachTo:rootNode:velocity:")]
 		SKAction ReachTo (CGPoint position, SKNode rootNode, nfloat velocity);
 
 		[iOS (8,0), Mac(10,10)]
 		[Static, Export ("reachToNode:rootNode:duration:")]
-		SKAction ReachToNode (SKNode node, SKNode rootNode, double sec);
+		SKAction ReachToNode (SKNode node, SKNode rootNode, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 31]
```diff
@@ -2117,27 +2154,27 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[iOS (8,0), Mac(10,10)]
 		[Static, Export ("reachToNode:rootNode:velocity:")]
 		SKAction ReachToNode (SKNode node, SKNode rootNode, nfloat velocity);
 		
 		[iOS (8,0), Mac (10,10)] // this method is missing the NS_AVAILABLE macro, but it shows up in the 10.10 sdk, but not the 10.9 sdk.
 		[Static, Export ("strengthTo:duration:")]
-		SKAction StrengthTo (float /* float, not CGFloat */ strength, double sec);
+		SKAction StrengthTo (float /* float, not CGFloat */ strength, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 32]
```diff
@@ -2117,27 +2154,27 @@ partial interface SKAction : NSCoding, NSCopying {
...
 
 		[iOS (8,0), Mac (10,10)] // this method is missing the NS_AVAILABLE macro, but it shows up in the 10.10 sdk, but not the 10.9 sdk.
 		[Static, Export ("strengthBy:duration:")]
-		SKAction StrengthBy (float /* float, not CGFloat */ strength, double sec);
+		SKAction StrengthBy (float /* float, not CGFloat */ strength, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 33]
```diff
@@ -2150,7 +2187,7 @@ partial interface SKAction : NSCoding, NSCopying {
 		[iOS (8,0), Mac(10,10)]
 		[Static]
 		[Export ("falloffTo:duration:")]
-		SKAction FalloffTo (float falloff, double sec);
+		SKAction FalloffTo (float falloff, double duration);
```
Comment (dalexsoto) [inline src/spritekit.cs]: same as above

File: src/spritekit.cs [hunk 34]
```diff
@@ -3223,5 +3260,80 @@ interface SKWarpGeometryGrid : NSCoding
...
+		SKRenderer FromDevice ([NullAllowed] IMTLDevice device);
+
+		[Export ("renderWithViewport:commandBuffer:renderPassDescriptor:")]
+		void Render (CGRect viewport, IMTLCommandBuffer commandBuffer, MTLRenderPassDescriptor renderPassDescriptor);
+
+		[Export ("renderWithViewport:renderCommandEncoder:renderPassDescriptor:commandQueue:")]
+		void Render (CGRect viewport, IMTLRenderCommandEncoder renderCommandEncoder, MTLRenderPassDescriptor renderPassDescriptor, IMTLCommandQueue commandQueue);
+
+		[Export ("updateAtTime:")]
+		void UpdateAtTime (double currentTime);
```
Comment (dalexsoto) [inline src/spritekit.cs]: Maybe just `Update`?

File: Make.config (hunk 1 above)
Comment (VincentDondain) [inline Make.config]: The message says that we can bump whenever we test with a more recent version and it works (:

Comment (dalexsoto) [inline Make.config]: Yeah, I completely agree when you are not in a release branch but I think that message is more intended for `master`

File: src/spritekit.cs (hunk 2 above)
Comment (VincentDondain) [inline src/spritekit.cs]: Mmh API diff considers this a breaking change. I don't really understand why it would be one...

File: src/spritekit.cs (hunk 34 above)
Comment (VincentDondain) [inline src/spritekit.cs]: Agreed. 👍

File: src/spritekit.cs (hunk 2 above)
Comment (dalexsoto) [inline src/spritekit.cs]: Because you can do 

`var action = MoveBy (deltaX: 1, deltaY: 2, sec: 100);`

If anyone had the above code you will break them

Comment (dalexsoto) [inline src/spritekit.cs]: But I still consider this is a good one, it really helps to better understand the API and even more if we add the measurement unit to the parameter name

Comment (VincentDondain) [inline src/spritekit.cs]: Aaaaaah you're right, I didn't think about that, I never name anything explicitly in C# :P

File: Make.config (hunk 1 above)
Comment (VincentDondain) [inline Make.config]: It's already not the same version as the d15.3 branch, I really think it doesn't matter: https://github.com/xamarin/xamarin-macios/blob/d15-3/Make.config#L63

Comment (dalexsoto) [inline Make.config]: Yeah, I think you changed that in the first Xcode9 bump https://github.com/xamarin/xamarin-macios/commit/24cd73a5ca280246b2778509b6846a3103d4e95d#diff-f682ea58f8badb13b9f8ed17cda4deb7R63 but yeah I think it does not matter much at this point unless we really want to be close to dev 15.3

File: runtime/bindings-generator.cs [hunk 35]
```diff
@@ -1444,6 +1444,26 @@ static IEnumerable<FunctionData> GetFunctionData ()
...
+					ReturnType = Types.QuatF,
+				}
+			);
+
+			data.Add (
+				new FunctionData {
+					Comment = " // void func (Quaternion)",
+					Prefix = "simd__",
+					Variants = Variants.NonStret,
+					Parameters = new ParameterData[] {
```
Comment (spouliot) [inline runtime/bindings-generator.cs]: minor: space before `[]`

File: src/spritekit.cs (hunk 2 above)
Comment (spouliot) [inline src/spritekit.cs]: It's a breakage change and it's less clear than (already less than perfect) `sec`, please keep the existing name.

File: tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs [hunk 36]
```diff
@@ -0,0 +1,72 @@
...
+				Assert.Inconclusive ("Requires Xcode9+");
+		}
+
+		[Test]
+		public void EulerAngles ()
+		{
+			Vector3 V3;
+
+			using (var obj = new SKTransformNode ())
+			{
```
Comment (spouliot) [inline tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs]: minor: should be on the previous line

File: tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs [hunk 37]
```diff
@@ -0,0 +1,72 @@
...
+				Assert.AreEqual (1.14159274f, obj.EulerAngles.Y, "#y1");
+				Assert.AreEqual (-0.141592711f, obj.EulerAngles.Z, "#z1");
+			}
+		}
+
+		[Test]
+		public void RotationMatrix ()
+		{
+			using (var obj = new SKTransformNode ())
+			{
```
Comment (spouliot) [inline tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs]: same

File: tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs [hunk 38]
```diff
@@ -0,0 +1,72 @@
...
+			}
+		}
+
+		[Test]
+		public void QuaternionTest ()
+		{
+			Quaternion Q;
+
+			using (var obj = new SKTransformNode ())
+			{
```
Comment (spouliot) [inline tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs]: same (the IDE should set it up properly, double check your settings and file a bug if set to mono style).

File: Make.config (hunk 1 above)
Comment (rolfbjarne) [inline Make.config]: The change itself is fine, but should go in a different PR, it has nothing to do with sprite kit :smile:

File: tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs [hunk 39]
```diff
@@ -0,0 +1,72 @@
...
+namespace MonoTouchFixtures.SpriteKit {
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class SKTransformNodeTest {
+		[SetUp]
+		public void VersionCheck ()
+		{
+			if (!TestRuntime.CheckXcodeVersion (9,0))
+				Assert.Inconclusive ("Requires Xcode9+");
```
Comment (rolfbjarne) [inline tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs]: minor (for next time): there's a one-liner for this:

```
TestRuntime.AssertXcodeVersion (9,0);
```

File: tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs [hunk 40]
```diff
@@ -0,0 +1,72 @@
...
+		public void EulerAngles ()
+		{
+			Vector3 V3;
+
+			using (var obj = new SKTransformNode ())
+			{
+				Asserts.AreEqual (Vector3.Zero, obj.EulerAngles, "1 EulerAngles");
+				V3 = new Vector3 (1, 2, 3);
+				obj.EulerAngles = V3;
+				Assert.AreEqual (-2.14159298f, obj.EulerAngles.X, "#x1");
```
Comment (rolfbjarne) [inline tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs]: You're setting a vector of [1,2,3] and getting something back that looks like random numbers. Are you sure the test isn't finding a bug somewhere?

File: tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs [hunk 41]
```diff
@@ -0,0 +1,72 @@
...
+			}
+		}
+
+		[Test]
+		public void RotationMatrix ()
+		{
+			using (var obj = new SKTransformNode ())
+			{
+				obj.RotationMatrix = Matrix3.Zero;
+				Asserts.AreEqual (Matrix3.Identity, obj.RotationMatrix, "RotationMatrix");
```
Comment (rolfbjarne) [inline tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs]: Also: setting `Matrix3.Zero`, and getting back `Matrix3.Identity`. This doesn't look random, but it still looks strange (so I'd test with Xcode, and added a comment about it).

File: tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs (hunk 40 above)
Comment (VincentDondain) [inline tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs]: Same values as native Swift example.

File: tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs (hunk 41 above)
Comment (VincentDondain) [inline tests/monotouch-test/SpriteKit/SKTransformNodeTest.cs]: Tested with Xcode already (found that weird too), same behavior. I'll add a comment that it does the same thing with native code.

File: runtime/bindings-generator.cs (hunk 35 above)
Comment (VincentDondain) [inline runtime/bindings-generator.cs]: 👍 Copied it from code above :P

**Comment by VincentDondain** [comment]
introspection and monotouch tests ran on device, both pass.

File: src/spritekit.cs [hunk 42]
```diff
@@ -3223,5 +3260,80 @@ interface SKWarpGeometryGrid : NSCoding
...
 		SKWarpGeometryGrid _GridByReplacingDestPositions (IntPtr destPositions);
 	}
+
+	// SKRenderer is not available for WatchKit apps and the iOS simulator
+	[NoWatch]
+	[TV (11,0), Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface SKRenderer
+	{
```
Comment (dalexsoto) [inline src/spritekit.cs]: Minor: Brace should be on the above line

File: src/spritekit.cs [hunk 43]
```diff
@@ -3223,5 +3260,80 @@ interface SKWarpGeometryGrid : NSCoding
...
+		bool ShowsPhysics { get; set; }
+
+		[Export ("showsFields")]
+		bool ShowsFields { get; set; }
+	}
+
+	[TV (11,0), Watch (4,0), Mac (13,0), iOS (11,0)]
+	[BaseType (typeof(SKNode))]
+	interface SKTransformNode
+	{
```
Comment (dalexsoto) [inline src/spritekit.cs]: Minor: Brace should be on the above line

**Comment by VincentDondain** [comment]
Unrelated test failure: https://bugzilla.xamarin.com/show_bug.cgi?id=55719
